src/AppBundle/Repository/PostRepository.php

```diff
@@ -38,4 +38,24 @@ public function findLatest($limit = Post::NUM_ITEMS)
             ->getResult()
         ;
     }
+
+    public function findByTerms(array $terms, $limit = Post::NUM_ITEMS)
+    {
+        $queryBuilder = $this->createQueryBuilder('p');
```

Comment:
its really a bad practice to get variable names under 4 characters so IMO that's definitely not something that should be encouraged in this project